docs/UserGuide.md

```diff
@@ -158,147 +128,241 @@ Examples:
   Parameters: [LIST] [KEYWORDS]...
   Example: list SCHEDULE
   Example: list STUDENTS phone email
-  Example: list TASKS`
+  Example: list TASKS
+```
+
+### Showing a lesson's/task's/student's details : `show`
+
+The show command has different behaviours depending on the current list:
+
+1. In `SCHEDULE` list:
+    - Shows the details of the specified lesson from the schedule list in the application. 
+2. In `TASKS` list:
+    - Shows the details of the specified task from the full task list in the application.
+3. In `STUDENTS` list:
+    - Shows the details of the specified student from the contact list in the application.
+
+Format: `show INDEX`
+
+* Shows the details of the lesson/task/student at the specified `INDEX`.
+* The index refers to the index number shown in the displayed schedule/task/student list.
+* The index **must be a positive integer** 1, 2, 3, …​
+
+Examples:
+1. In `SCHEDULE` list:
+    *  `list SCHEDULE` followed by `show 2` shows the details of the 2nd lesson in the schedule list.
+    * `find lesson1` followed by `show 1` shows the details of the 1st lesson in the results of the `find` command.
+2. In `TASKS` list:
+    *  `list TASKS` followed by `show 2` shows the description of the 2nd task in the full task list.
+3. In `STUDENTS` list:
+    * `list STUDENTS` followed by `show 2` shows the details of the 2nd student in the student list.
+    * `find Betsy` followed by `show 1` shows the details of the 1st student in the results of the `find` command.
+
+Success Output:
+* In `SCHEDULE` list:
+```
+Showing Lesson: Start: 2:30 PM; End: 3:00 PM
+```
+* In `TASKS` list:
 ```
+Showing Task: Description: Revise CS2103T Materials
+```
+* In `STUDENTS` list:
+```
+Showing Person: Alex Yeoh; Phone: 87438807; Email: alexyeoh@example.com; Address: Blk 30 Geylang Street 29, #06-40; Subjects: [BIOLOGY]; Tags: [friends]
+```
+
+Failure Output:
+```  
+  Invalid command format!
+  show: Shows the details of the item identified by the index number used in the last item listing.
+  Parameters: INDEX (must be a positive integer)
+  Example: show 1
+ ```
+
+### Adding a student : `addPerson`
+
+Adds a student to the contact list in application.
+
+Format: `addPerson -name NAME [-phone PHONE_NUMBER] [-email EMAIL] [-address ADDRESS] 
+[-subject SUBJECT] [-tag TAG] [-remark REMARK]`
+
+<box type="tip" seamless>
 
-### Editing a student : `edit`
+**Tips:** 
+- A student can have any number of unique tags (including 0)
+- A new student cannot have the same name as existing students in the contact list.
+- If the user is currently in list `STUDENTS`, the command can be shortened to `add`
+- For flags that can take multiple values (eg. -subject, -tag), separate the values with commas
```

Comment:
Someone might want to abuse this. Could you help make it clear that tags cannot have non-alpha number characters in them (example, "-", spaces and ","), and tags and subjects must be unique? thanks

Reply:
Made the necessary changes in recent commits. Thank you!